### DISCUSSION TITLE
Provide entropy when generating a keypair?

### DISCUSSION THREAD

**Discussion by AndreasGassmann** [openpgpjs/openpgpjs#1309] [Q&A]
[BIP85](https://github.com/bitcoin/bips/blob/master/bip-0085.mediawiki#rsa-gpg) describes a way how entropy can be derived deterministically from a BIP32 root key, for example a BIP39 mnemonic (24 words) commonly used in cryptocurrency wallets.

In our app, [AirGap Vault](https://github.com/airgap-it/airgap-vault), we would like to add the functionality to let users derive GPG keys from their seeds using BIP85. The python reference implementation includes an example how it is done: https://github.com/ethankosakovsky/bip85/blob/7892bbddc51f498d1a693cca3f6d642b75cb3373/bip85/tests/test_bip85rsa.py

The problem is that I can't see a way to provide my own entropy/randomness when generating a new keypair. I quickly went through the code and I don't see an option to do that, unless I missed it. I assume somewhere here the entropy would have to be provided instead of the actual random buffer: https://github.com/openpgpjs/openpgpjs/blob/93b77669bc24512198c6f0b49f19fe617ed5f1cc/src/crypto/public_key/elliptic/curves.js#L193

Is there a way how entropy can be provided to the generateKeypair method? If not, would there be a straight forward and safe way to somehow generate the keypair by using some internal methods?

**Comment by larabr** [comment]
This is not something we support, and [neither does WebCrypto](https://stackoverflow.com/questions/55464963/how-do-you-set-a-seed-or-entropy-or-default-string-for-keys-generated-by-web). Related discussion: https://github.com/openpgpjs/openpgpjs/discussions/1245#discussioncomment-395002 .
